vscode-lean4/src/leanclient.ts

```diff
@@ -378,9 +377,19 @@ export class LeanClient implements Disposable {
 
     isSameWorkspace(uri: Uri){
         if (this.folderUri) {
-            if (uri.toString().startsWith(this.folderUri.toString())) {
-                // skip it, this file belongs to a different workspace...
-                return true;
+            if (this.folderUri.scheme !== uri.scheme) return false;
+            if (this.folderUri.scheme === 'file') {
+                const realPath1 = fs.realpathSync(this.folderUri.fsPath).toLowerCase();
+                const realPath2 = fs.realpathSync(uri.fsPath).toLowerCase();
+                if (process.platform === 'win32') {
+                    // windows paths are case insensitive.
+                    return realPath2.startsWith(realPath1);
+                } else {
+                    return realPath2.toString().startsWith(realPath1);
```

Comment:
This `toString` is a no-op since `realPath2` is already a string.

Reply:
fixed